services/repository/template.go

```diff
@@ -16,8 +16,29 @@ import (
 	repo_module "code.gitea.io/gitea/modules/repository"
 )
 
+// GenerateIssueLabels generates issue labels from a template repository
+func GenerateIssueLabels(ctx context.Context, templateRepo, generateRepo *repo_model.Repository) error {
+	templateLabels, err := models.GetLabelsByRepoID(ctx, templateRepo.ID, "", db.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, templateLabel := range templateLabels {
+		generateLabel := &models.Label{
+			RepoID:      generateRepo.ID,
+			Name:        templateLabel.Name,
+			Description: templateLabel.Description,
+			Color:       templateLabel.Color,
+		}
+		if err := db.Insert(ctx, generateLabel); err != nil {
+			return err
+		}
+	}
```

Comment:
```suggestion
	newLabels := make([]*models.Label, 0, len(templateLabels))
	for _, templateLabel := range templateLabels {
		newLabels = append(newLabels, &models.Label{
			RepoID:      generateRepo.ID,
			Name:        templateLabel.Name,
			Description: templateLabel.Description,
			Color:       templateLabel.Color,
		})
	}
	if err := db.Insert(ctx, newLabels); err != nil {
		return err
	}
```
Should do the same, needs a bit more memory but enables a batch insert.